README.md

```diff
@@ -53,26 +53,57 @@ For details on all functionality of this library, see the [GoDoc][godoc] documen
 
 Below are a few simple examples:
 
+### Read Secrets
 ```go
+package main
+
+import (
+    "fmt"
+
+    "github.com/secrethub/secrethub-go/pkg/secrethub"
+)
+
+func main() {
+    client, _ := secrethub.NewClient()
+    secret, _ := client.Secrets().ReadString("path/to/db/pass")
+    fmt.Println(secret)
+    // Output: wFc16W#96N1$
+}
+```
+
+### Write Secrets
+```go
+package main
+
 import (
-	"github.com/secrethub/secrethub-go/pkg/randchar"
-	"github.com/secrethub/secrethub-go/pkg/secrethub"
+    "fmt"
+
+    "github.com/secrethub/secrethub-go/pkg/secrethub"
 )
 
-// Setup
-credential, err := secrethub.NewCredential("<your credential>", "<passphrase>")
-client := secrethub.NewClient(credential, nil)
+func main() {
+    client, _ := secrethub.NewClient()
+    _, _ = client.Secrets().Write("path/to/secret", []byte("password123"))
+}
+```
 
-// Write
-secret, err := client.Secrets().Write("path/to/secret", []byte("password123"))
+### Generate Secrets
+```go
+package main
 
-// Read
-secret, err = client.Secrets().Versions().GetWithData("path/to/secret:latest")
-fmt.Println(secret.Data) // prints password123
+import (
+    "fmt"
+
+    "github.com/secrethub/secrethub-go/pkg/randchar"
+    "github.com/secrethub/secrethub-go/pkg/secrethub"
+)
 
-// Generate a slice of 32 alphanumeric characters.
-data, err := randchar.NewGenerator(false).Generate(32)
-secret, err = client.Secrets().Write("path/to/secret", data)
+func main() {
+    client, _ := secrethub.NewClient()
+    rand, _ := randchar.NewRand(randchar.Alphanumeric)
+    data, _ := rand.Generate(30)
```

Comment:
Can we simplify this even further to take the default rand? E.g. I'd like to see utility functions on the package that use a `randchar.Default` to make super easy one-liners e.g. `randchar.Generate(30)`. Alternatively, `randchar.Default.Generate(30` also works.

Reply:
Because the underlying OS random number generator can return an error. We could make it so it doesn't return an error and panics instead, but I don't like that so much.

Reply:
Ah gotcha, yes NewRand could not return an error, but options currently return an error. There are some valid cases where it should, I think, e.g. when you add mutually exclusive options

Reply:
We can update the example when we have this shortcut default functionality.

Reply:
https://github.com/secrethub/secrethub-go/pull/142